### PR TITLE
LPS-172418 Relationship Tab is not exported properly

### PR DESCRIPTION
**Pull request by marcelabc**
Related issue: https://liferay.atlassian.net/browse/LPS-172418

cc: @seltong @leilanyaragao

### CODE REVIEW

**Comment by marcelabc** [comment]
Fixing rebase conflict.

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectLayoutResourceImpl.java [hunk 1]
```diff
@@ -306,7 +324,7 @@ private com.liferay.object.model.ObjectLayoutTab _toObjectLayoutTab(
 				objectLayoutBox -> _toObjectLayoutBox(
 					objectDefinitionId, objectLayoutBox)));
 		serviceBuilderObjectLayoutTab.setObjectRelationshipId(
-			GetterUtil.getLong(objectLayoutTab.getObjectRelationshipId()));
+			objectRelationshipId);
```
Comment (carolmariaabb) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectLayoutResourceImpl.java:327]: As used, i.e., declare `objectRelationshipId` as close as possible to its first line of usage:

```
long objectRelationshipId = GetterUtil.getLong(
	objectLayoutTab.getObjectRelationshipId());

if (Validator.isNotNull(
	objectLayoutTab.getObjectRelationshipExternalReferenceCode())) {

	ObjectRelationship objectRelationship =
		_objectRelationshipLocalService.
			getObjectRelationshipByExternalReferenceCode(
				objectLayoutTab.
					getObjectRelationshipExternalReferenceCode(),
				contextCompany.getCompanyId(), objectDefinitionId);

	objectRelationshipId = objectRelationship.getObjectRelationshipId();
}

serviceBuilderObjectLayoutTab.setObjectRelationshipId(
	objectRelationshipId);
```

OR

```
if (Validator.isNotNull(
		objectLayoutTab.getObjectRelationshipExternalReferenceCode())) {

	ObjectRelationship objectRelationship =
		_objectRelationshipLocalService.
			getObjectRelationshipByExternalReferenceCode(
				objectLayoutTab.
					getObjectRelationshipExternalReferenceCode(),
				contextCompany.getCompanyId(), objectDefinitionId);

	serviceBuilderObjectLayoutTab.setObjectRelationshipId(
		objectRelationship.getObjectRelationshipId());
}
else {
	serviceBuilderObjectLayoutTab.setObjectRelationshipId(
		GetterUtil.getLong(objectLayoutTab.getObjectRelationshipId()));
}
```

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java [hunk 2]
```diff
@@ -799,8 +810,9 @@ public ObjectRelationship updateObjectRelationship(
...
 
 		if (objectRelationship.isReverse()) {
-			throw new ObjectRelationshipReverseException(
-				"Reverse object relationships cannot be updated");
+			return objectRelationshipLocalService.
+				updateReverseObjectRelationship(
+					externalReferenceCode, objectRelationship);
 		}
 
 		_validateExternalReferenceCode(
```
Comment (carolmariaabb) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java:818]: This new method `updateReverseObjectRelationship` is only used here, so we could remove it and change the code to:

```
_validateExternalReferenceCode(
	externalReferenceCode, objectRelationshipId,
	objectRelationship.getCompanyId(),
	objectRelationship.getObjectDefinitionId1());

if (objectRelationship.isReverse()) {
	objectRelationship.setExternalReferenceCode(externalReferenceCode);

	return objectRelationshipPersistence.update(objectRelationship);
}

_validateParameterObjectFieldId(
```

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by marcelabc** [comment]
ci:test:sf

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectLayoutUtil.java [hunk 3]
```diff
@@ -68,12 +72,17 @@ public static ObjectLayout toObjectLayout(
...
+		ObjectRelationshipLocalService objectRelationshipLocalService) {
 
 		if (objectLayoutTab == null) {
 			return null;
 		}
 
+		ObjectRelationship objectRelationship =
+			objectRelationshipLocalService.fetchObjectRelationship(
+				objectLayoutTab.getObjectRelationshipId());
+
```
Comment (carolmariaabb) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectLayoutUtil.java:85]: @marcelabc is this code necessary? You already did this [here](https://github.com/liferay-objects/liferay-portal/blob/ad02ab4dce86add0cc281e35c9ebf15b53562b3a/modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectLayoutUtil.java#L102).

**Comment by marcelabc** [comment]
ci:test:sf

**Review by carolmariaabb — APPROVED** [review]

**Comment by carolmariaabb** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
ci:forward:force

**Pull request closed without merging**